### PR TITLE
Avoid overriding virtual host names provided by the user

### PR DESCRIPTION
**Pull request by glyn**
[#127696153]

### CODE REVIEW

**Comment by glyn** [comment]
This code has not been changed since it was reviewed by @scottfrederick.

File: spring-cloud-services-spring-connector/src/main/java/io/pivotal/spring/cloud/service/eureka/EurekaInstanceAutoConfiguration.java [hunk 1]
```diff
@@ -69,6 +69,12 @@
 	@Value("${spring.cloud.services.registrationMethod:route}")
 	private String registrationMethod;
 
+	@Value("${virtual-host-name:#{null}}")
+	private String providedVirtualHostname;
+
+	@Value("${secure-virtual-host-name:#{null}}")
+	private String providedSecureVirtualHostname;
+
```
Comment (royclarkson) [inline spring-cloud-services-spring-connector/src/main/java/io/pivotal/spring/cloud/service/eureka/EurekaInstanceAutoConfiguration.java]: It's a larger change, but we should consider using a properties file, rather than field injection.

Comment (glyn) [inline spring-cloud-services-spring-connector/src/main/java/io/pivotal/spring/cloud/service/eureka/EurekaInstanceAutoConfiguration.java]: It's a fair cop governor! I would like to avoid field injection too. @royclarkson has kindly agreed to raise a separate story to avoid scope creep here.
